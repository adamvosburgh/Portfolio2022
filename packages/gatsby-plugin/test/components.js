import React from 'react'
import renderer from 'react-test-renderer'
import {
  Notes,
  Head,
  Header,
  Footer,
  Color,
  Invert,
  Steps,
  Image,
  Horizontal,
  Split,
  SplitRight,
  FullScreenCode,
} from '../src/components'

const render = el => renderer.create(el).toJSON()

test('Color renders', () => {
  const json = render(
    <Color bg='tomato' />
  )
  expect(json).toMatchSnapshot()
})

test('Invert renders', () => {
  const json = render(
    <Invert />
  )
  expect(json).toMatchSnapshot()
})

test('Steps renders', () => {
  const json = render(
    <Steps>
      <div>One</div>
      <div>Two</div>
    </Steps>
  )
  expect(json).toMatchSnapshot()
})

test('Steps renders with list', () => {
  const json = render(
    <Steps>
      <ul originalType='ul'>
        <li>One</li>
        <li>Two</li>
      </ul>
    </Steps>
  )
  expect(json).toMatchSnapshot()
})

test('Image renders', () => {
  const json = render(
    <Image src='kittens.png' />
  )
  expect(json).toMatchSnapshot()
})

test('Horizontal renders', () => {
  const json = render(
    <Horizontal>
      <div>A</div>
      <div>B</div>
    </Horizontal>
  )
  expect(json).toMatchSnapshot()
})

test('Split renders', () => {
  const json = render(
    <Split>
      <div>A</div>
      <div>B</div>
    </Split>
  )
  expect(json).toMatchSnapshot()
})

test('SplitRight renders', () => {
  const json = render(
    <SplitRight>
      <div>A</div>
      <div>B</div>
    </SplitRight>
  )
  expect(json).toMatchSnapshot()
})

test('FullScreenCode renders', () => {
  const json = render(
    <FullScreenCode>
      Hi
    </FullScreenCode>
  )
  expect(json).toMatchSnapshot()
})

