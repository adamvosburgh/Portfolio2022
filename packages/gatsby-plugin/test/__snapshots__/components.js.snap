// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color renders 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  background-color: tomato;
}

.emotion-0 a {
  color: inherit;
}

<div
  className="emotion-0"
/>
`;

exports[`FullScreenCode renders 1`] = `
.emotion-0 {
  width: 100%;
  height: 100%;
}

.emotion-0 pre {
  margin: 0 !important;
  width: 100%;
  height: 100%;
  overflow: auto;
}

<div
  className="emotion-0"
>
  Hi
</div>
`;

exports[`Horizontal renders 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  text-align: center;
}

.emotion-0 {
  width: 50%;
}

.emotion-0 img {
  height: auto;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0"
  >
    <div>
      A
    </div>
  </div>
  <div
    className="emotion-0"
  >
    <div>
      B
    </div>
  </div>
</div>
`;

exports[`Image renders 1`] = `
.emotion-0 {
  width: 100%;
  height: 100%;
  background-size: cover;
  background-image: url(kittens.png);
  background-position: center;
  background-repeat: no-repeat;
}

<div
  className="emotion-0"
/>
`;

exports[`Invert renders 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  color: background;
  background-color: text;
}

.emotion-0 a {
  color: inherit;
}

<div
  className="emotion-0"
/>
`;

exports[`Split renders 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  text-align: center;
}

.emotion-0 {
  width: 50%;
}

.emotion-0 img {
  height: auto;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0"
  >
    <div>
      A
    </div>
  </div>
  <div
    className="emotion-0"
  >
    <div>
      B
    </div>
  </div>
</div>
`;

exports[`SplitRight renders 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  text-align: center;
}

.emotion-0 {
  width: 50%;
}

.emotion-0 img {
  height: auto;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0"
  >
    <div>
      B
    </div>
  </div>
  <div
    className="emotion-0"
  >
    <div>
      A
    </div>
  </div>
</div>
`;

exports[`Steps renders 1`] = `
Array [
  <div
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    One
  </div>,
  <div
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    Two
  </div>,
]
`;

exports[`Steps renders with list 1`] = `
<ul
  originalType="ul"
>
  <li
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    One
  </li>
  <li
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    Two
  </li>
</ul>
`;
